RecoTracker/LSTCore/interface/alpaka/Constants.h

```diff
@@ -0,0 +1,126 @@
+    using ::lst::Params_T3;
+    using ::lst::Params_T5;
+    using ::lst::size_superbins;
+
+// Half precision wrapper functions.
+#if defined(FP16_Base)
+#define __F2H __float2half
+#define __H2F __half2float
+    typedef __half float FPX;
+#else
+#define __F2H
+#define __H2F
+    typedef float FPX;
+#endif
+
+    Vec3D constexpr elementsPerThread(Vec3D::all(static_cast<Idx>(1)));
+
+// Needed for files that are compiled by g++ to not throw an error.
+// uint4 is defined only for CUDA, so we will have to revisit this soon when running on other backends.
+#if !defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    struct uint4 {
+      unsigned int x;
+      unsigned int y;
+      unsigned int z;
+      unsigned int w;
+    };
+#endif
+
+    // Adjust grid and block sizes based on backend configuration
+    template <typename Vec, typename TAcc = Acc<typename Vec::Dim>>
+    ALPAKA_FN_HOST ALPAKA_FN_INLINE WorkDiv<typename Vec::Dim> createWorkDiv(const Vec& blocksPerGrid,
+                                                                             const Vec& threadsPerBlock,
+                                                                             const Vec& elementsPerThreadArg) {
+      Vec adjustedBlocks = blocksPerGrid;
+      Vec adjustedThreads = threadsPerBlock;
+
+      // special overrides for CPU/host cases
+      if constexpr (std::is_same_v<Platform, alpaka::PlatformCpu>) {
+        adjustedBlocks = Vec::all(static_cast<Idx>(1));
+
+        if constexpr (alpaka::accMatchesTags<TAcc, alpaka::TagCpuSerial>) {
+          // Serial execution, set threads to 1 as well
+          adjustedThreads = Vec::all(static_cast<Idx>(1));  // probably redundant
+        }
+      }
+
+      return WorkDiv<typename Vec::Dim>(adjustedBlocks, adjustedThreads, elementsPerThreadArg);
+    }
+
+    // The constants below are usually used in functions like alpaka::math::min(),
+    // expecting a reference (T const&) in the arguments. Hence,
+    // ALPAKA_STATIC_ACC_MEM_GLOBAL needs to be used in addition to constexpr.
+
+    // 15 MeV constant from the approximate Bethe-Bloch formula
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float kMulsInGeV = 0.015;
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float kMiniMulsPtScaleBarrel[6] = {
+        0.0052, 0.0038, 0.0034, 0.0034, 0.0032, 0.0034};
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float kMiniMulsPtScaleEndcap[5] = {0.006, 0.006, 0.006, 0.006, 0.006};
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float kMiniRminMeanBarrel[6] = {
+        25.007152356, 37.2186993757, 52.3104270826, 68.6658656666, 85.9770373007, 108.301772384};
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float kMiniRminMeanEndcap[5] = {
+        130.992832231, 154.813883559, 185.352604327, 221.635123002, 265.022076742};
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float k2Rinv1GeVf = (2.99792458e-3 * 3.8) / 2;
+    ALPAKA_STATIC_ACC_MEM_GLOBAL constexpr float kR1GeVf = 1. / (2.99792458e-3 * 3.8);
```

Comment:
IIRC, there is no support in CMS for continuous magnetic field values; only a set (smth like 4, 3.8, 2, 0) are supported. This code will not work for 0. It can be realistic to consider another value (2 T) at some point in not too distant future.

Having some samples and a related validation campaign cycle would help. It would be good to understand what are the realistic goals of this: long term physics running or interim 1-2 fills "with just some kind of meaningful tracking". If I recall the past situations, 0 T is a more likely long term problematic situation.

For physics, to the zeroth order the LST code should be still functional if the field is e.g. at 2 T: almost all of the code is geometric (except for may be pixel seed/tracks where p_T is used), the output of LST to CMSSW is a pattern of hits which is converted to a track state aware of a varying magnetic field. So, a change in the field can be traded for a change in the nominal momentum cutoff, e.g. from 0.8 to `3.8/2*0.8 = 1.5`.
Higher order computational and physics performance tuning will be needed to get things more perfect for 2 T.